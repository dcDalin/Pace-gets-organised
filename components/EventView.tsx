import NewEventModal from "@/components/NewEventModal";
import DayCard from "@/components/DayCard";

interface IEventViewProps {
  eventsLayout: any[];
  handleOpenNewEventModal: (state: boolean) => void;
  openNewEventModal: boolean;
}
export default function EventView({
  eventsLayout,
  handleOpenNewEventModal,
  openNewEventModal,
}: IEventViewProps) {
  return (
    <>
      <div
        className="h-[1440px] w-[620px] px-0 box-border bg-gray-50 cursor-pointer"
        onClick={() => handleOpenNewEventModal(true)}
        data-testid="events-view"
      >
        {eventsLayout &&
          eventsLayout.length &&
          eventsLayout.map((event, index) => {
            return <DayCard key={index} event={event} />;
          })}
      </div>

      <NewEventModal
        open={openNewEventModal}
        close={() => handleOpenNewEventModal(false)}
      />
    </>
  );
}
