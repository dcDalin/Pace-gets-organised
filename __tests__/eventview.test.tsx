import { render, screen, waitFor } from "@testing-library/react";
import EventView from "@/components/EventView";
import "@testing-library/jest-dom";
import { act } from "react-dom/test-utils";

const observe = jest.fn();

describe("Event View", () => {
  it("renders event view unchanged", () => {
    const { container } = render(
      <EventView
        eventsLayout={[]}
        handleOpenNewEventModal={() => {}}
        openNewEventModal={false}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("opens new event modal", async () => {
    render(
      <EventView
        eventsLayout={[]}
        handleOpenNewEventModal={jest.fn()}
        openNewEventModal={false}
      />
    );

    screen.getByTestId("events-view").click();

    const newEventModal = await waitFor(() =>
      screen.getByTestId("new-event-modal")
    );

    expect(newEventModal).toBeInTheDocument();
  });

  it("finds title of event", () => {
    act(async () => {
      const { getByTestId } = render(
        <EventView
          eventsLayout={[
            {
              end: 440,
              height: 20,
              id: "1",
              left: 110,
              start: 420,
              title: "Call with Bob",
              top: 420,
              units: 2,
              width: 300,
            },
          ]}
          handleOpenNewEventModal={() => {}}
          openNewEventModal={false}
        />
      );

      await waitFor(() => {
        const title = getByTestId("event-title");
        expect(title.innerHTML).toBe("Call with Bob");
        console.log("Title: ", title);
      });
    });
  });
});
