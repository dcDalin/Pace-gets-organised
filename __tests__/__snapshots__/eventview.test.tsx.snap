// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event View renders event view unchanged 1`] = `
<div>
  <div
    class="h-[1440px] w-[620px] px-0 box-border bg-gray-50 cursor-pointer"
    data-testid="events-view"
  >
    0
  </div>
</div>
`;
